src/lib.rs

```diff
@@ -216,14 +247,20 @@ impl<K: Eq + Hash, V, S: BuildHasher> LruCache<K, V, S> {
     /// assert_eq!(cache.get_mut(&2), None);
     /// assert_eq!(cache.get_mut(&3), Some(&mut "c"));
     /// ```
-    pub fn set_capacity(&mut self, capacity: usize) {
+    pub fn set_capacity(&mut self, capacity: usize) -> Vec<(K, V)> {
+        let cap = capacity.saturating_sub(self.len());
+        let mut removed_lrus = Vec::with_capacity(cap);
+
         for _ in capacity..self.len() {
-            self.remove_lru();
+            let removed = self.remove_lru().unwrap();
+            removed_lrus.push(removed);
         }
```

Comment:
Would using an iterator to create `removed_lrus` be as space efficient as your capacity specific vector?

I don't know how the efficiency compares, but I prefer reading the iterator version:

```rust
let removed_lrus = capacity..self.len()
    .filter_map(|_|: self.remove_lru())
    .collect();
```